code/headers/data_extractor.hpp

```diff
@@ -0,0 +1,53 @@
+#pragma once
+#include <cstdint>
+
+namespace r2d2::thermal_camera {
+    class data_extractor_s {
+    public:
+        // static class
+        data_extractor_s() = delete;
+
```

Comment:
Why is this implemented as a deleted class and not just a namespace with functions? (docs would help)